### PR TITLE
Folding in editor, code generation improvements

### PR DESCRIPTION
**Pull request by hza**
folding 
Screenshot 2023-07-24 at 20 49 16

### CODE REVIEW

File: src/main/java/ui/dialog/TextEditorComponent.java [hunk 1]
```diff
@@ -23,55 +31,106 @@ public interface CodeProvider<T, R> {
...
         EditorFactory editorFactory = EditorFactory.getInstance();
-        Document document = editorFactory.createDocument(code);
 
-        // Get the file type
         FileTypeManager fileTypeManager = FileTypeManager.getInstance();
         FileType fileType = fileTypeManager.getFileTypeByExtension(settings.getFormat());
 
-        // Create the editor with syntax support
-        editor = (EditorEx) editorFactory.createEditor(document, project, fileType, false);
+        String content = code.replace("\r\n", "\n");
```
Comment (hza) [inline src/main/java/ui/dialog/TextEditorComponent.java:61]: if some problems on windows with encoding it is better to fix it here

File: src/main/java/action/CopyValueAction.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ protected void perform(XValueNodeImpl node, @NotNull String nodeName, AnActionEv
                 if (isExtractorPlugin && settings.getFormat().equals("json")) {
                     String jsonString = JsonCodeProvider.genJsonString(node);
                     if (settings.isPrettyFormat()) {
-                        jsonString = JacksonUtil.prettyFormatWithoutSlashR(jsonString);
+                        jsonString = JsonUtil.formatJson(node.getTree().getProject(), jsonString);
```
Comment (hza) [inline src/main/java/action/CopyValueAction.java:44]: replacing formatting

File: src/main/java/util/debugger/ValueJsonSerializer.java [hunk 3]
```diff
@@ -188,12 +180,6 @@ private static boolean isSimpleValue(Value value) {
         }
     }
 
-    @NotNull
-    private static String toValRefString(ObjectReference valRef) {
```
Comment (chocovon) [inline src/main/java/util/debugger/ValueJsonSerializer.java:192]: I tried it using that Tester's case I mentioned before, it resulted like this:
```
{
  ""{
  dvsltest.Tester@16b4a017=dvsltest.Tester@16b4a017,
  dvsltest.Tester@424c0bc4=dvsltest.Tester@3c679bde,
  dvsltest.Tester@3c679bde=dvsltest.Tester@424c0bc4
}"":{"name": "t3", "type": "hh", "obj": {"name": "t1", "type": "tt","obj": null}}, ""{dvsltest.Tester@16b4a017=dvsltest.Tester@16b4a017, dvsltest.Tester@424c0bc4=dvsltest.Tester@3c679bde, dvsltest.Tester@3c679bde=dvsltest.Tester@424c0bc4}"": {"name": "t2", "type": "pp", "obj": null},""{dvsltest.Tester@16b4a017=dvsltest.Tester@16b4a017, dvsltest.Tester@424c0bc4=dvsltest.Tester@3c679bde, dvsltest.Tester@3c679bde=dvsltest.Tester@424c0bc4}"":null
}
```
which is not a valid json string.

Comment (hza) [reply src/main/java/util/debugger/ValueJsonSerializer.java:192]: fixed in [314eedd](https://github.com/chocovon/debug-variable-sl/pull/5/commits/314eedda6b063f01bd548d61decaa7f41a50977f)

now names are like in jackson library.

Screenshot 2023-07-25 at 19 54 07

**Comment by hza** [comment]
added filtering of this$

**Comment by hza** [comment]
added 'Hide null values' and max depth option to dialog

**Comment by hza** [comment]
added folding for java code too

Screenshot 2023-07-26 at 22 51 55

File: inner-tool/src/main/java/util/code/ObjectCodeHelper.java
Comment (chocovon) [inline inner-tool/src/main/java/util/code/ObjectCodeHelper.java:1]: Are there some reason to do this refactoring? For me it looks like it exposes inner complexity of the Helper class, and the "final" won't make sure the set is immutable. It seems to be better to make them private and only provide easy-to-use methods for external usage.

Comment (hza) [reply inner-tool/src/main/java/util/code/ObjectCodeHelper.java:1]: fixed

**Comment by hza** [comment]
fresh screenshots:


Screenshot 2023-07-27 at 21 42 11
Screenshot 2023-07-27 at 21 42 16

**Pull request merged**